components/style/properties/longhand/box.mako.rs

```diff
@@ -1926,6 +1926,201 @@ ${helpers.single_keyword("transform-style",
     }
 </%helpers:longhand>
 
+<%helpers:longhand name="contain" animatable="False" products="gecko"
+                   spec="https://drafts.csswg.org/css-contain/#contain-property">
+    use std::fmt;
+    use style_traits::ToCss;
+    use values::HasViewportPercentage;
+
+    no_viewport_percentage!(SpecifiedValue);
+
+    pub mod computed_value {
+        #[derive(Clone, Debug, PartialEq)]
+        #[cfg_attr(feature = "servo", derive(HeapSizeOf))]
+        pub enum ComputedContainment {
+            Size,
+            Layout,
+            Style,
+            Paint,
+        }
+
+        #[derive(Clone, Debug, PartialEq)]
+        #[cfg_attr(feature = "servo", derive(HeapSizeOf))]
+        pub struct T(pub Option<Vec<ComputedContainment>>);
```

Comment:
Indeed, the bitflags can just be something like:

```rust
bitflags! {
    pub flags StyleContain : u8 {
        PAINT = 0x01,
        LAYOUT = 0x02,
        STYLE = 0x04,
        SIZE = 0x08,
    }
```

Then we could have some convenience functions for `strict`, etc, and serialization should test from more to less broad.

That makes `to_computed_value` a no-op (and unnecessary).

WDYT?